### ISSUE TITLE
Collect benchmarks and improve FFI performance

### ISSUE THREAD

**Issue by dcharkes** [dart-lang/sdk#36247]
- [x] Benchmarks on Golem
  - [x] benchmarks
    - [x] micro benchmark memory loads and stores
    - [x] micro benchmark struct loads and stores
    - [x] micro benchmark ffi trampoline calls
    - [ ] micro benchmark ffi callbacks (moved to #38171)
    - [x] macro benchmark (some boringssl or alike)
  - [x] architectures
    - [x] x64
    - [x] ia32
    - [x] arm64
    - [x] arm (benchmarks work, but hardfp not supported yet  #36309)
    - [x] simdbc64 on x64
- [x] Design doc `Pointer.store` and `Pointer.load` in IL
- [ ] Implement `Pointer.store` and `Pointer.load` in IL (moved to #38172)

**Comment by sjindel-google** [comment]
https://github.com/santetis/libsodium.dart contains a crypto benchmark

**Comment by dcharkes** [comment]
Note to self. Loop invariant code motion should garbage collect the Pointer object too early, as that would call finalizers too early. See TypedData, which has the same issue.

**Comment by sjindel-google** [comment]
To scope this for the MVP release: we will understand the performance of FFI operations and implement any "low-hanging" optimisations.

**Comment by dcharkes** [comment]
Arm32 benchmarks on Linux are blocked by https://github.com/dart-lang/sdk/issues/36309.

**Comment by sjindel-google** [comment]
We need benchmarks for FFI callbacks also.

**Comment by dcharkes** [comment]
Closing in favor of the new issues.
